src/treemodel/selection.js

```diff
@@ -229,44 +229,49 @@ export default class Selection {
 	/**
 	 * Sets collapsed selection in the specified location.
 	 *
-	 * The location can be specified as:
+	 * The location can be specified in the same form as {@link engine.treeModel.Position.createAt} parameters.
 	 *
-	 * * a {@link engine.treeModel.Position position},
-	 * * parent element and offset (offset defaults to `0`),
-	 * * parent element and `'END'` (sets selection at the end of that element),
-	 * * node and `'BEFORE'` or `'AFTER'` (sets selection before or after the given node).
-	 *
-	 * @fires {@link engine.treeModel.Selection#change:range change:range}
+	 * @fires engine.treeModel.Selection#change:range
 	 * @param {engine.treeModel.Node|engine.treeModel.Position} nodeOrPosition
 	 * @param {Number|'END'|'BEFORE'|'AFTER'} [offset=0] Offset or one of the flags. Used only when
 	 * first parameter is a node.
 	 */
 	collapse( nodeOrPosition, offset ) {
-		let node, pos;
+		const pos = Position.createAt( nodeOrPosition, offset );
+		const range = new Range( pos, pos );
 
-		if ( nodeOrPosition instanceof Position ) {
-			pos = nodeOrPosition;
-		} else {
-			node = nodeOrPosition;
-
-			if ( offset == 'END' ) {
-				offset = node.getChildCount();
-			} else if ( offset == 'BEFORE' ) {
-				offset = node.getIndex();
-				node = node.parent;
-			} else if ( offset == 'AFTER' ) {
-				offset = node.getIndex() + 1;
-				node = node.parent;
-			} else if ( !offset ) {
-				offset = 0;
-			}
+		this.setRanges( [ range ] );
+	}
+
+	/**
+	 * Sets {@link engine.treeModel.Selection#focus} in the specified location.
+	 *
+	 * The location can be specified in the same form as {@link engine.treeModel.Position.createAt} parameters.
+	 *
+	 * @fires engine.treeModel.Selection#change:range
+	 * @param {engine.treeModel.Node|engine.treeModel.Position} nodeOrPosition
+	 * @param {Number|'END'|'BEFORE'|'AFTER'} [offset=0] Offset or one of the flags. Used only when
+	 * first parameter is a node.
+	 */
+	setFocus( nodeOrPosition, offset ) {
```

Comment:
The result of this function may be a range that intersects with other ranges currently in seleciton. AFAIR at one point we said that this should not happen. This behavior may have a negative influence on some scripts (i.e. when a scripts takes all ranges from selection and processes them).